tsdb/mapper.go

```diff
@@ -39,7 +39,8 @@ func (mo *mapperOutput) key() string {
 // LocalMapper is for retrieving data for a query, from a given shard.
 type LocalMapper struct {
 	shard           *Shard
-	stmt            *influxql.SelectStatement
+	stmt            influxql.Statement
```

Comment:
Then an instance of this struct becomes an attribute of `LocalMapper`. I think this will do a lot for keeping the code manageable.